thoth/user_api/api_v1.py

```diff
@@ -394,6 +396,111 @@ def list_python_package_indexes():
     return graph.python_package_index_listing()
 
 
+def post_build(
+    build_detail: dict,
+    debug: bool = False,
+    registry_user: str = None,
+    registry_password: str = None,
+    environment_type: str = None,
+    origin: str = None,
+    registry_verify_tls: bool = True,
+    force: bool = False,
+):
+    """Run analysis on a build."""
+    response = {"base_image_analysis": {}, "output_image_analysis": {}, "build_log_analysis": {}}
+    status = 202
+    if build_detail.get("output_image"):
+        # Run image analysis
+        output_image_analyze_response, output_image_analyze_status = post_analyze(
+            image=build_detail["output_image"],
+            debug=debug,
+            registry_user=registry_user,
+            registry_password=registry_password,
+            environment_type=environment_type,
+            origin=origin,
+            verify_tls=registry_verify_tls,
+            force=force,
+        )
+        response["output_image_analysis"] = output_image_analyze_response
+        if output_image_analyze_status != 202:
+            return response, output_image_analyze_status
+
+    if build_detail.get("base_image"):
+        # Run base image analysis
+        base_image_analyze_response, base_image_analyze_status = post_analyze(
+            image=build_detail["base_image"],
+            debug=debug,
+            environment_type=environment_type,
+            origin=origin,
+            verify_tls=registry_verify_tls,
+            force=force,
+        )
+        response["base_image_analysis"] = base_image_analyze_response
+        if base_image_analyze_status != 202:
+            return response, base_image_analyze_status
+
+    if build_detail.get("log"):
+        # attach image analysis details to build log
+        if build_detail.get("output_image") and output_image_analyze_response:
+            build_detail["output_image_analysis_id"] = output_image_analyze_response["analysis_id"]
+        if build_detail.get("base_image") and base_image_analyze_response:
+            build_detail["base_image_analysis_id"] = base_image_analyze_response["analysis_id"]
+
+        # Run build log analysis
+        buildlog_analyze_response, buildlog_analyze_status = post_buildlog_analyze(log_info=build_detail, force=force)
+        response["build_log_analysis"] = buildlog_analyze_response
+        if buildlog_analyze_status != 202:
+            return response, buildlog_analyze_status
+
+    return response, status
```

Comment:
If None of the three was provided? There should be returned HTTP status code 400 signalizing a bad request.